YK's final content review feedback (#22)
OpenLiberty/draft-guide-okd: README.adoc

:common-includes: https://raw.githubusercontent.com/OpenLiberty/guides-common/master
:source-highlighter: prettify
:page-seo-title: Deploying Java microservices to an OKD cluster using Minishift
:page-seo-description: A getting started tutorial and an example on how to deploy Java microservices to an Origin Community Distribution of Kubernetes (OKD) cluster using Minishift for development purposes.
:guide-author: Open Liberty
= Deploying microservices to an OKD cluster using Minishift

[.hidden]
NOTE: This repository contains the guide documentation source. To view the guide in published form, view it on the https://openliberty.io/guides/{projectid}.html[Open Liberty website^].

Explore how to use Minishift to deploy microservices to an Origin Community Distribution of Kubernetes (OKD) cluster. 

// *********************************************************************************************

== What you'll learn 

You will learn how to deploy two simple microservices with Open Liberty to an Origin Community Distribution of Kubernetes (OKD) cluster that is running in 
Minishift.

=== What is Origin Community Distribution of Kubernetes (OKD)?

> OKD, formerly known as OpenShift Origin, is the upstream open source project for all OpenShift products. OKD is a Kubernetes-based platform with

Reviewer: ```suggestion
OKD, formerly known as OpenShift Origin, is the upstream open source project for all RedHat OpenShift products. OKD is a Kubernetes-based platform with
```

added functionality. OKD streamlines the DevOps process by providing an intuitive development pipeline. It also provides integration with multiple tools 
to make the deployment and management of cloud applications easier.

To learn more about the different platforms that Red Hat OpenShift offers, check out the 
https://docs.openshift.com[official OpenShift documentation^]. To learn how to deploy microservices to OpenShift Online, check out the 
https://openliberty.io/guides/cloud-openshift.html[Deploying microservices to OpenShift^] guide. If you would like to learn more about Kubernetes, check out 

> the https://openliberty.io/guides/kubernetes-intro.html[Deploying microservices to Kubernetes^] guide.

Reviewer: We should update the links above to point to OKD instead?  Not sure we need links to OpenShift?  We can keep the link to the Kubernetes guide.

mvn -pl system package
```

The `mvn package` command compiles, verifies, and builds the project. The resulting compiled code is packaged into a `war` web archive that can be 
found under the `system/target` directory. The archive contains the application that is needed to run the microservice on an Open Liberty server, and it is 
now ready to be injected into a Docker container for deployment.

== Containerizing the system microservice

=== Reusing the Docker daemon

> To speed up the development process, the built-in Minishift Docker daemon can be reused to avoid having to build a Docker registry on the host machine. 

Reviewer: ```suggestion
To simplify the local deployment process, the built-in Minishift Docker daemon can be reused to avoid having to build a Docker registry on the host machine. 
```

mvn -pl system package
```

The `mvn package` command compiles, verifies, and builds the project. The resulting compiled code is packaged into a `war` web archive that can be 
found under the `system/target` directory. The archive contains the application that is needed to run the microservice on an Open Liberty server, and it is 
now ready to be injected into a Docker container for deployment.

== Containerizing the system microservice

=== Reusing the Docker daemon

To simplify the local deployment process, the built-in Minishift Docker daemon can be reused. Reusing the Minishift Docker daemon allows your system to 
connect to it, and use the internal Docker registry. This avoids having to build a Docker registry on your machine. To reuse the Docker daemon, run the 

> following command:

Reviewer: ```suggestion
following command to point your shell to use the Minishift's daemon:
```

include::{common-includes}/os-tabs.adoc[]
[.tab_content.windows_section]
--
Run the following command, noting the values of the `system` and `inventory` route hostnames:

[role=command]
```
oc get routes
```

> Substitute `[system-route-hostname]` and `[inventory-route-hostname]` with the appropriate values in the following command:

Reviewer: ```suggestion
Substitute `[system-route-hostname]` and `[inventory-route-hostname]` with the appropriate values and run the following command:
```

Author: The values are taken from the command that follows, the change implies that the substitution should occur before running the command that displays the substitution values.

Reviewer: I do not understand this.  If it is what you are saying -- i.e. run command to obtain values, substitute and then, execute another command -- we shall rework the section.

Author: My mistake, I thought this was a previous section. You're right, committing the change

oc delete -f deploy.yaml
```

To completely delete your Minishift VM, cluster, and all associated files, refer to the official 
https://docs.okd.io/latest/minishift/getting-started/uninstalling.html[Uninstalling Minishift^] documentation.

> To revert your Docker daemon back to its default settings, simply close the shell session.

Reviewer: ```suggestion
To revert your Minishift Docker daemon back to its default settings, simply close the shell session.
```

Author: The guide asks the user to allow their system to use Minishift's daemon, so technically we're not reverting the Minishift daemon back to the original settings, instead, we're disconnecting our console from the built-in Minishift Docker daemon and allowing our system to revert back to using our Docker daemon.

Reviewer: Then, let us explain that appropriately.  Both before and after.  It wasn't clear still what we are asking the users to do.  Your explanation here is much better than what we have.

oc delete -f deploy.yaml
```

To completely delete your Minishift VM, cluster, and all associated files, refer to the official 
https://docs.okd.io/latest/minishift/getting-started/uninstalling.html[Uninstalling Minishift^] documentation.

> To disconnect from Minishift's internal Docker daemon and reconnect to your local daemon, simply close your shell session.

Reviewer: There isn't actual disconnection and reconnection, right?  Closing the shell will get rid of the environment settings and starting a new one will give you your default ones?
